aws/provider.go

```diff
@@ -1098,6 +1098,7 @@ func Provider() *schema.Provider {
 			"aws_ssoadmin_permission_set_inline_policy":               resourceAwsSsoAdminPermissionSetInlinePolicy(),
 			"aws_storagegateway_cache":                                resourceAwsStorageGatewayCache(),
 			"aws_storagegateway_cached_iscsi_volume":                  resourceAwsStorageGatewayCachedIscsiVolume(),
+			"aws_storagegateway_fsx_associate_file_system":            resourceAwsStorageGatewayFsxAssociateFileSystem(),
```

Comment:
I should have caught this earlier, but the resource names should be nouns or noun phrases. Something like `aws_storagegateway_file_system_association`, based on the API call names `DescribeFileSystemAssociations` and `UpdateFileSystemAssociation`.

Reply:
so we chose this naming based on: 
[AssociateFileSystem](https://docs.aws.amazon.com/storagegateway/latest/APIReference/API_AssociateFileSystem.html) is the API call to create and we prepended the `fsx` to add clarity and based on the other storagegateway resources, ex:

[CreateNFSFileShare](https://docs.aws.amazon.com/storagegateway/latest/APIReference/API_CreateNFSFileShare.html) is named `resource_aws_storagegateway_nfs_file_share` already.

if we were to change it, id argue we should just remove `FSx` however, i personally think its a good addition for clarity. please lmk what you think, @gdavison